pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,149 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+
+	// Provider contains settings for different providers for which an agent based cluster installation
+	// can be performed.
+	Provider Provider `json:"provider"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	// TODO: should this be optional and defaulted?
+	Networking Networking `json:"networking"`
+
+	// UserManagedNetworking indicates if the networking is managed by the user.
+	// TODO: more detail needed here.
+	// +optional
+	UserManagedNetworking bool `json:"userManagedNetworking,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to all cluster hosts.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// SSHPublicKey will be added to all cluster hosts for use in debugging.
+	// +optional
+	SSHPublicKey string `json:"sshPublicKey,omitempty"`
+}
+
+// Provider contains settings for different providers for which an agent based cluster installation
+// can be performed.
+type Provider struct {
+	// BareMetal contains bare metal specific configuration for this cluster.
+	BareMetal BareMetal `json:"bareMetal"` // mandatory for now, possibly optional in future if other platforms are supported
+}
+
+// BareMetal defined agent based install configuration specific to bare metal clusters.
+type BareMetal struct {
+	// APIVIP is the virtual IP used to reach the OpenShift cluster's API.
+	APIVIP string `json:"apiVIP"`
+	// APIVIPDNSName is the domain name used to reach the OpenShift cluster API.
+	APIVIPDNSName string `json:"apiVIPDNSName"`
+	// IngressVIP is the virtual IP used for cluster ingress traffic.
+	IngressVIP string `json:"ingressVIP"`
+	// VIPDHCPAllocation indicates if virtual IP DHCP allocation mode is enabled.
+	VIPDHCPAllocation bool `json:"vipDHCPAllocation"`
+}
+
+// Networking defines the pod network provider in the cluster.
+type Networking struct {
+	// NetworkType is the type of network to install. The default is OpenShiftSDN
+	//
+	// +kubebuilder:default=OpenShiftSDN
+	// +optional
+	//NetworkType string `json:"networkType,omitempty"`
```

Comment:
Is this commented out on purpose?

Reply:
Yes I was meaning to ask if this needs to be exposed in the first draft.

Reply:
Ok then for now lets drop from API and we'll keep that assumption internal for time being. We're looking at possibly taking an optional install config template elsewhere as an input which would cover this use case if it ever needed to be explicit.